piwebapi_samples/Data Analysis/PIUploadUtility/PIUploadUtility/PIUploadUtility/Building Example.xml

```diff
@@ -0,0 +1,9095 @@
+        <Name>Outside Relative Humidity at VAV Start</Name>
+        <DefaultUOM>%</DefaultUOM>
+        <Type>Double</Type>
+        <Value type="Double">0</Value>
+        <DataReference>PI Point</DataReference>
+        <ConfigString>\\localhost\Lab Building Data\Building\Weather|Relative Humidity percent;TimeRangeMethod=StartTime</ConfigString>
+        <AFAttributeCategoryRef>Start Time Values</AFAttributeCategoryRef>
+      </AFAttributeTemplate>
+      <AFAttributeTemplate>
+        <Name>Setpoint Offset at start time</Name>
+        <Type>Double</Type>
+        <Value type="Double">0</Value>
+        <DataReference>Formula</DataReference>
+        <ConfigString>R=Room Temperature at VAV Start;S=Setpoint at VAV Start;[R-S]</ConfigString>
+        <AFAttributeCategoryRef>Start Time Values</AFAttributeCategoryRef>
+      </AFAttributeTemplate>
+    </AFElementTemplate>
+    <AFElement>
+      <Name>Building</Name>
+      <AFElement ReferenceType="Parent-Child">
+        <Name>Floor_2</Name>
+        <Template>Floor</Template>
+        <AFAttribute>
+          <Name>Device ID</Name>
+          <Value type="Double">1012</Value>
+          <AFAttributeCategoryRef>Asset Information</AFAttributeCategoryRef>
+        </AFAttribute>
+        <AFAttribute>
+          <Name>Floor</Name>
+          <ConfigString>Right(%Element%,1);</ConfigString>
+          <AFAttributeCategoryRef>Asset Information</AFAttributeCategoryRef>
+        </AFAttribute>
+        <AFElement ReferenceType="Parent-Child">
+          <Name>VAVCO 2-03</Name>
+          <Description>Device 101215</Description>
+          <Template>VAVCO</Template>
+          <AFAttribute>
+            <Name>% cooling</Name>
+            <DataReference>PI Point</DataReference>
+            <ConfigString>\\localhost\Building_Data_101215.AnalogValue1.PresentValue</ConfigString>
+            <AFAttributeCategoryRef>Control</AFAttributeCategoryRef>
+            <AFAttributeCategoryRef>Cooling</AFAttributeCategoryRef>
+          </AFAttribute>
+          <AFAttribute>
+            <Name>AC Unit</Name>
+            <Value type="String">AC-2</Value>
+            <AFAttributeCategoryRef>Metadata</AFAttributeCategoryRef>
+          </AFAttribute>
+          <AFAttribute>
+            <Name>Device ID</Name>
+            <Value type="Double">101215</Value>
+            <AFAttributeCategoryRef>Metadata</AFAttributeCategoryRef>
+          </AFAttribute>
+          <AFAttribute>
+            <Name>Predicted Cooling Time</Name>
+            <DataReference>PI Point</DataReference>
+            <ConfigString>\\localhost\VAVCO 2-03.Predicted Cooling Time</ConfigString>
+            <AFAttributeCategoryRef>Forecasts</AFAttributeCategoryRef>
+          </AFAttribute>
+          <AFAttribute>
+            <Name>Setpoint Offset Current</Name>
+            <DataReference>PI Point</DataReference>
+            <ConfigString>\\localhost\VAVCO 2-03.Setpoint Offset Current.1315f488-6e35-5fba-1531-384ac098e1ef</ConfigString>
```

Comment:
PI Point paths in the AF attribute assume a local PI DA.  One options would be to add a comment in the instructions that this XML file should be edited to point to the correct PI DA.

Reply:
Looking further, I think the problem is that the default data server for the newly created AF DB is not specified to the one in "test_config.json".  In an effort to resolve the template, which does use %Server%, it falls back to localserver.